### PR TITLE
Add updates for CH4 analytical inversion

### PR DESCRIPTION
**Pull request by msulprizio**
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard 

### Describe the update

This pull request includes two major changes to the CH4 simulation:

**1) Move CH4 emissions handling out of GEOS-Chem code and into HEMCO_Config.rc**

Code for handling CH4 emissions has been removed from both global_ch4_mod.F90 and carbon_mod.F90. Previously, emissions from all sectors were combined into the State_Chm%CH4_EMIS (or CH4_EMIS_J in carbon_mod.F90). This was done so that total emissions could be perturbed for use in analytical inversions. With this update, the emissions are now perturbed within HEMCO_Config.rc using scale factors defined for each state vector element (read from a netCDF file). The perturbations are meant to be used on total CH4 emissions (i.e. using output from a HEMCO standalone simulation) to avoid having to apply the scale factor(s) for every emissions inventory and sector throughout HEMCO_Config.rc. A new logical switch UseTotalPriorEmis has been added to HEMCO_Config.rc for this purpose, though users will need to modify the file path to point to their prior emissions.

For example:

```
#==============================================================================
# ---Total CH4 emissions (all sectors) from prior simulation ---
#==============================================================================
(((UseTotalPriorEmis
0 CH4_Emis_Prior ../prior_run/OutputDir/HEMCO_diagnostics.$YYYY$MM$DD0000.nc EmisCH4_Total $YYYY/$MM/$DD/0 C xy kg/m2/s CH4 - 1 500
0 CH4_Emis_Prior_0001 - - - - - - CH4_0001 2001 1 500
)))UseTotalPriorEmis

#==============================================================================
# --- Scale factors for analytical inversions ---
#==============================================================================
(((AnalyticalInversion
# Add perturbations to individual state vector element (N) following the format
# Start scale factor ID at 2000 to avoid conflicts with other SFs/masks
#200N SCALE_ELEM000N ./Perturbations.txt - - - xy count 1
2001 SCALE_ELEM_0001 ./Perturbations.txt - - - xy count 1
)))AnalyticalInversion
```

The Emis_PosteriorSF and OH_PosteriorSF options have also been updated so they are also applied directly within HEMCO_Config.rc instead of within global_ch4_mod.F90/carbon_mod.F90. Emis_PosteriorSF applies the field EMIS_SF to the total CH4 emissions field and OH_PosteriorSF applies OH_SF to the GLOBAL_OH field.

Several options have also been removed from the CH4 menu in geoschem_config.yml since the were used in code that was removed from global_ch4_mod.F90.

**2) Allow for multiple CH4 tracers**

Routine CH4_DECAY in global_ch4_mod.F90 has been modified to loop over all advected species instead of assuming there is only one CH4 tracer. This allow components of the state vector to be added as individual CH4 tracers (e.g. CH4_0001, CH4_0002) so that jacobian simulations within the Integrated Methane Inversion (IMI) framework can be combined into fewer runs. The IMI automatically updates the geoschem_config.rc and species database files for the additional tracers so that does not need to be done by GEOS-Chem.

### Expected changes

This is a zero-difference update with respect to the full-chemistry benchmark simulation.

### Related Github Issue(s)

- https://github.com/geoschem/integrated_methane_inversion/pull/193
- https://github.com/geoschem/integrated_methane_inversion/issues/119
- https://github.com/geoschem/integrated_methane_inversion/issues/120

### CODE REVIEW

**Review by yantosca — CHANGES_REQUESTED** [review]
Thanks @msulprizio.  I noted a couple of minor things to be updated.

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Copy utility scripts that allow you to resubmit failed to integration and parallel test root directories
 - Update GCHP operational example environment files for Harvard Cannon
 - Do not run GCClassic integration test compile jobs in the background
+- Removed emissions handling from `global_ch4_mod.F90` and `carbon_mod.F90` and instead apply scale factors to emissions directly in `HEMCO_Config.rc`
```
[the remark this answers is not in the document]
Comment (msulprizio) [reply CHANGELOG.md]: Now fixed in https://github.com/geoschem/geos-chem/commit/75a886bb47b7500522570c09df352f44c2d46c37

File: GeosCore/input_mod.F90 [hunk 2]
```diff
@@ -4983,46 +4983,6 @@ SUBROUTINE Config_CH4( Config, Input_Opt, RC )
     ENDIF
     Input_Opt%TCCON_CH4_OBS = v_bool
 
-    !------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (msulprizio) [reply GeosCore/input_mod.F90:4986]: Thanks for pointing that out. I will submit a separate PR in the GCClassic repository for the doc updates.

[the remark this answers is not in the document]
Comment (msulprizio) [reply GeosCore/input_mod.F90:4986]: See https://github.com/geoschem/GCClassic/pull/53 which should be merged into dev/no-diff-to-benchmark alongside this pull request.

[the remark this answers is not in the document]
Comment (yantosca) [reply GeosCore/input_mod.F90:4986]: @msulprizio: I approved https://github.com/geoschem/GCClassic/pull/53 and will bring it into 14.3.1.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (yantosca) [reply CHANGELOG.md]: Thanks!

**Review by yantosca — APPROVED** [review]
All remaining items have now been addressed.  Good to merge.  Thank you @msulprizio!

**Comment by yantosca** [comment]
All GEOS_Chem Classic integration tests passed:
```console
==============================================================================
GEOS-Chem Classic: Execution Test Results

GCClassic #50d8dd0 GEOS-Chem submod update: Merge PR #2199 (GCHP advection diags)
GEOS-Chem #afd1f596b Merge PR #2218 (Add updates for CH4 analytical inversion)
HEMCO     #2a4bb12 Merge PR #268 (Handle 3D NEI emissions)

Using 24 OpenMP threads
Number of execution tests: 26

Submitted as SLURM job: 25565699
==============================================================================

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
````
All GEOS-Chem Classic integration tests were zero-diff w/r/t 14.3.0 except:
- APM (known parallelization issue)
- RRTMG (numerical noise in RRTMG collection file)
- CH4, tagCH4 (updates related to this PR)

**Comment by yantosca** [comment]
All GCHP integration tests passed:
```console
==============================================================================
GCHP: Execution Test Results

GCHP      #4b6eb6a FvDyCoreCubed_GridComp update: Merge PR #8 (Add PLEadv diagnostic)
GEOS-Chem #afd1f596b Merge PR #2218 (Add updates for CH4 analytical inversion)
HEMCO     #2a4bb12 Merge PR #268 (Handle 3D NEI emissions)

Number of execution tests: 6

Submitted as SLURM job: 25565991
==============================================================================
 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%  All execution tests passed!  %%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Also, all GCHP integration tests were zero-diff w/r/t 14.3.0.
